### ISSUE TITLE
Bug: Misleading message for non-configured files

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#18263]
### Environment

Node version: v20.12.0
npm version: v10.5.0
Local ESLint version: v9.0.0-rc.0
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = [];
```
</details>

```js
// index.ts
```

Command:

```
eslint index.ts
```

### What did you expect to happen?

This file doesn't have a matching configuration, so I'd expect a message describing that.

### What actually happened?

```
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to disable file ignore settings or use "--no-warn-ignored" to suppress this warning
```

This message is misleading since the file is not ignored "because of a matching ignore pattern". Also, using "--no-ignore" wouldn't help.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-zxcj6m

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
I agree. We should have a better error message.

**Comment by fasttime** [comment]
I've prepared a PR to show how to improve the error message in the specific case that a file is missing a matching configuration, it's marked as a draft because it requires updating `config-array` first: #18404.

**Comment by spamshaker** [comment]
what is the default matcher for files? I also experienced the same issue for .jsx extensions
